source/agora/node/Ledger.d

```diff
@@ -227,31 +254,43 @@ public class Ledger
 
     public Hash[] getMerklePath (ulong block_height, Hash hash) @safe
     {
-        if (block_height >= this.ledger.length)
+        if (this.block_height < block_height)
             return null;
 
-        const(Block)* block = &this.ledger[block_height];
-
-        size_t index = block.findHashIndex(hash);
-        if (index < block.txs.length)
-            return block.getMerklePath(index);
+        Block block;
+        if (this.storage.readBlock(block, block_height))
```

Comment:
@TrustHenry : From experience, returning early leads to more readable code.
For example:
```d
if (!this.storage.readBlock(block, block_height))
    assert(0);
size_t index = block.findHashIndex(hash);
if (index >= block.txs.length)
    return null;
return block.getMerklePath(index);
```

In this example, we return early when there's an error, so it is easier for the reader to follow the flow.

Reply:
Applied the suggestion